typing/typeopt.ml

```diff
@@ -122,6 +122,7 @@ let type_sort ~why env loc ty =
    so this type is polymorphic in what it remembers about products. *)
 type 'a classification =
   | Int   (* any immediate type *)
+  | Immediate_or_null
```

Comment:
Since the "Int" in the case above actually meants immediate, I think that either this constructor should be called "Int_or_null" or the other one should be renamed "Immediate".